core/shared/communication/base-communication.ts

```diff
@@ -32,7 +44,17 @@ export abstract class BaseCommunication {
     });
   }
 
-  protected onResponse(message: Message): void {
+  protected async onMessageReceived(message: Message): Promise<void> {
+    const actOn = this.subscribers.get(message.operation);
+    const answer = await actOn?.(message);
+    if (answer) {
+      await this.send(answer);
+    } else {
+      this.onResponse(message);
+    }
```

Comment:
This means, if the subscriber returns undefined, we will treat it as a response 🤔

Reply:
I would probably revert it. If we have a pending request with the same ID, it's a response. Else let's try if we have a subscriber.